### ISSUE TITLE
LoRA loss is going NaN after 1st epoch - Kohya SS GUI

### ISSUE THREAD

**Issue by shubhamworks** [bmaltais/kohya_ss#1112]
I've been trying a style with LoRA. I've 82 input images, with 5 times repeated. And I'm training 8 epochs with arch size = 2. After the first epoch, my training loss is going NaN, and a corrupted safetensors file generating. Any suggestions?

**Comment by nothelloearth1** [comment]
What dimension/alpha/learning rate (text encoder and unet) are you using?

**Comment by jsnavely** [comment]
I'm getting the same issue.  I don't update Kohya SS very often, but did an update last week to try out the LoHA method Bernard showed in a YouTube video.  The LoHA trained without issue, though the results did not look as good as my regular LoRA version from the same dataset.  I did notice that during training the LoHA loss rate hovered back and forth around 0.1 or 0.11 for every epoch and never dropped.  I later tried to train LoRAs a couple times using the identical settings I have used for 20+ LoRAs and got NaN each time, so I used the LoHA method instead and they worked.  Today, I tried updating again to see if the problem was fixed and this time even trying to train as LoHA instead of normal LoRA is giving me NaN.  Either there is something wrong in the code, or my installation is messed up.  In my case, it keeps going to NaN before the 1st epoch completes, so I don't get any usable files.

**Comment by jsnavely** [comment]
The one difference between the LoHA I tried earlier today and the ones I did earlier in the week is that I changed gradient accumulation to 0.  I am now running it again with gradient accumulation set to 50 (image count / batch size) to see if it works this time.  So far I'm at 2% and the loss is around 0.102.  Somewhere around 1-2% is where it was going to NaN before, so it looks promising.

**Comment by jsnavely** [comment]
I tried making a new version of a standard LoRA that I had made in the past with additional images and concepts.  Using the same settings and expanded dataset gave me the NaN error, but then I used the new grouping utility to group the images by size (I'm using bucketing), changed the image directory to the grouped images directory, and the training worked.  One commonality of all of the LoRAs (standard, LoHA or SDXL) that I attempted over the last few weeks since updating is that I think all of the ones with ungrouped image directories failed with NaN, but all of the ones using grouped images were able to train without going to NaN.
